src/variable.cpp

```diff
@@ -77,6 +87,36 @@ void Variable::swap(Variable &rhs)
     std::swap(this->mPimpl, rhs.mPimpl);
 }
 
+void Variable::addEquivalence(const VariablePtr &equivalentVariable)
+{
+    setEquivalentTo(equivalentVariable);
+    VariablePtr *pV = new VariablePtr(this);
```

Comment:
A static method does indeed seem to get around the issue nicely. It also takes some of the possible ambiguity about reciprocity for anyone not digging into the documentation (I would expect reciprocity out of `addEquivalence(v1,v2)` but not necesarilly out of `v1.addEquivalence(v2)`)